docs/public/components/creating-components.md

```diff
@@ -58,27 +52,32 @@ This file hierarchy aimed at solving a few problems:
 * A main that imports a component should be able to select a specific implementation without compiling with the others.
   For example: the ZSTD library should not be included at compile time when the ZIP version is used.
 
-!!! warning "Important"
-    For all these reasons, components should never be nested.
 
-## Defining Components
+**Go package naming convention**:
 
-You can use the [invoke](../setup.md#invoke) task `inv components.new-component comp/<bundleName>/<component>` to generate a scaffold for your new component.
+All implementations must use the package name: `<component name>impl`.
+
+For example, a compression component with 2 implementation would use `package compressionimpl` in both
+`comp/<bundle>/compression/impl-zstd` and `comp/<bundle>/compression/impl-zip` folders.
+
+## Bootstraping Components
+
+You can use the [invoke](../setup.md#invoke) task `inv components.new-component comp/<component>` to generate a scaffold for your new component.
 
 Every public variable, function, struct, and interface of your component **must** be documented. Please refer to the [Documentation](#documentation) section below for details.
 
 ### The def folder
 
 The def folder will contain your interface and ALL public types needed by the users of your component.
 
-For our compression example is will look like this:
+For our compression example it will look like this:
 
-=== ":octicons-file-code-16: comp/&lt;bundleName&gt;/compression/def/component.go"
```

Comment:
I removed the `&lt;bundleName&gt;` to avoid adding cognitive overload to the readers. Since at this stage of the documentation they do not know about bundles

Reply:
true, but I think bundle are mandatory and people might copy past the the path,